### PR TITLE
Add Tree Render and Collision

### PR DESCRIPTION
**Pull request by ProgrammingIncluded**
## Collision
- Add Collision Layers (`collidableLayer`). Layers must be different in order to register a hit.
- Add Collidable flags for Entity which must be toggled for `onHit` to work
- Entities now have reference to `boardTree` instead and should us `addEntity` to add entities.
- On hit will be called when sprite touches another sprite. Both sprite's onhit will be called.

## Tree Rendering
- Board Tree renders each region as a chunk
   - Sandbox right now has chunk set at 256 which is tiny enough for you to see the culling.
   - Increase 256 to something like 728 to be larger than camera
- Entities will only update if you are on the same chunk or closest adjacent chunk
   - Unless you are on the edges, in which case some chunks don't render at all

## Font Rendering

- Removed WebFontLoader and instead uses native CSS loading to remove server issues
  -  also standalone builds unblocked for custom font.

## TODO:
- Add Camera movement

### CODE REVIEW

File: src/entity/player.js [hunk 1]
```diff
@@ -16,12 +16,24 @@ class Player extends Entity  {
...
+        this.boardTree = undefined;
+    }
+
+    fireBullet() {
+        this.boardTree.addEntity(
+            "bullet",
+            this.position.x,
+            this.position.y,
+            this.maxSpeed,
+            this.container.rotation + Math.PI, false
```
Comment (hirudan) [inline src/entity/player.js]: Small note here: I think we want this to be `this.container.rotation` and then the appropriate offset gets applied in `Bullet.load`. It was only `Math.PI` for my sample enemy shooting logic.

File: src/entity/player.js [hunk 2]
```diff
@@ -16,12 +16,24 @@ class Player extends Entity  {
...
 
         window.addEventListener('keydown', this.onKeyPress.bind(this));
         window.addEventListener('keyup', this.onKeyUp.bind(this));
 
         // set during load
-        this.board = undefined;
+        this.boardTree = undefined;
+    }
+
+    fireBullet() {
```
Comment (hirudan) [inline src/entity/player.js:29]: Looking below, I don't think we're calling this in `Player.attack`

**Review by hirudan — COMMENTED** [review]
Couple of questions, but overall this is really cool!

File: src/entity/player.js (hunk 1 above)
Comment (ProgrammingIncluded) [reply src/entity/player.js]: Updated, thx.

File: src/entity/player.js (hunk 2 above)
Comment (ProgrammingIncluded) [reply src/entity/player.js:29]: Thx!

**Review by hirudan — APPROVED** [review]

**Pull request merged**